src/rcc.rs

```diff
@@ -456,27 +486,33 @@ impl CFGR {
             while rcc.csr.read().lsirdy().bit_is_clear() {}
         }
 
-        // Turn on HSI48 if required
-        if self.hsi48 {
+        // Turn on USB, RNG Clock using the HSI48CLK source (default)
+        if !cfg!(feature = "stm32l47x") && self.usb48 {
             // p. 180 in ref-manual
             rcc.crrcr.modify(|_, w| w.hsi48on().set_bit());
             // Wait until HSI48 is running
             while rcc.crrcr.read().hsi48rdy().bit_is_clear() {}
         }
 
-
-
+        if cfg!(feature = "stm32l47x") && self.usb48 {
```

Comment:
So for MSI selection, I was thinking about maybe an Enum that has all possible pre configurable speeds, or taking the speed from an `Into<Hertz>` value, this would allow future users to use the MSI easily.

Reply:
Yes that would be intresting to have it. We need to solve how to enforce 48Mh if MSI is selected for the USB, RNG and SDMMC clock source.  The enum seems a better choise as there is a fix set of speed that can be selected.

Reply:
I think your right, the enum method makes sense. Using the enum allows peripherals easily to check what speed the MSI is running at.